cli-plugins/manager/candidate_test.go

```diff
@@ -80,7 +79,8 @@ func TestValidateCandidate(t *testing.T) {
 				assert.ErrorContains(t, err, tc.err)
 			case tc.invalid != "":
 				assert.NilError(t, err)
-				assert.Assert(t, cmp.ErrorType(p.Err, reflect.TypeOf(&pluginError{})))
+				var expectedError *pluginError
+				assert.Check(t, errors.As(p.Err, &expectedError))
```

Comment:
Oh! I recall some fun cases where there's no _good_ replacement, or at least `errors.As` / `errors.Is` for sure has a bunch of pitfalls; see the discussion I had with the gotest.tools authors on https://github.com/gotestyourself/gotest.tools/issues/272

Reply:
It's really horrible, and very easy to get wrong (in which case, it's even possible to get the unexpected result). I honestly really hate the `errors.As` / `errors.Is` handling for that reason.